dandi/tests/test_validate.py

```diff
@@ -39,9 +68,15 @@ def test_validate_bids_errors(bids_error_examples, dataset):
     # ideally make a list and erode etc.
 
     selected_dataset = os.path.join(bids_error_examples, dataset)
-    validation_result = validate_bids(selected_dataset, report=True)
+    validation_result = validate(selected_dataset)
     with open(os.path.join(selected_dataset, ".ERRORS.json")) as f:
         expected_errors = json.load(f)
+    validation_result = [i for i in validation_result]
```

Comment:
This line accomplishes nothing but wasted cycles.

Reply:
@jwodder I had hoped so too, but since we're now using the unified validation function with the `yield` return, it's actually needed:

```
(dev) [deco]~/src/dandi-cli/dandi ❱ pytest -vvs tests/test_validate.py::test_validate_bids_errors
=========================================================== test session starts ============================================================
platform linux -- Python 3.10.10, pytest-7.2.1, pluggy-1.0.0 -- /usr/bin/python3.10
cachedir: .pytest_cache
rootdir: /home/chymera/src/dandi-cli, configfile: tox.ini
plugins: httpserver-1.0.6, mock-3.10.0, pkgcore-0.12.19
collected 3 items

tests/test_validate.py::test_validate_bids_errors[invalid_asl003] Cloning into '/tmp/pytest-of-chymera/pytest-361/gitrepo0'...
remote: Enumerating objects: 145, done.
remote: Counting objects: 100% (145/145), done.
remote: Compressing objects: 100% (121/121), done.
remote: Total 145 (delta 20), reused 136 (delta 20), pack-reused 0
Receiving objects: 100% (145/145), 363.77 KiB | 3.06 MiB/s, done.
Resolving deltas: 100% (20/20), done.
FAILED
tests/test_validate.py::test_validate_bids_errors[invalid_eeg_cbm] FAILED
tests/test_validate.py::test_validate_bids_errors[invalid_pet001] FAILED

================================================================= FAILURES =================================================================
________________________________________________ test_validate_bids_errors[invalid_asl003] _________________________________________________
tests/test_validate.py:76: in test_validate_bids_errors
    assert len(validation_result) > 0
E   TypeError: object of type 'generator' has no len()
________________________________________________ test_validate_bids_errors[invalid_eeg_cbm] ________________________________________________
tests/test_validate.py:76: in test_validate_bids_errors
    assert len(validation_result) > 0
E   TypeError: object of type 'generator' has no len()
________________________________________________ test_validate_bids_errors[invalid_pet001] _________________________________________________
tests/test_validate.py:76: in test_validate_bids_errors
    assert len(validation_result) > 0
E   TypeError: object of type 'generator' has no len()
=========================================================== slowest 10 durations ===========================================================
0.61s setup    dandi/tests/test_validate.py::test_validate_bids_errors[invalid_asl003]
0.00s setup    dandi/tests/test_validate.py::test_validate_bids_errors[invalid_eeg_cbm]
0.00s teardown dandi/tests/test_validate.py::test_validate_bids_errors[invalid_pet001]
0.00s setup    dandi/tests/test_validate.py::test_validate_bids_errors[invalid_pet001]
0.00s call     dandi/tests/test_validate.py::test_validate_bids_errors[invalid_asl003]
0.00s teardown dandi/tests/test_validate.py::test_validate_bids_errors[invalid_asl003]
0.00s teardown dandi/tests/test_validate.py::test_validate_bids_errors[invalid_eeg_cbm]
0.00s call     dandi/tests/test_validate.py::test_validate_bids_errors[invalid_eeg_cbm]
0.00s call     dandi/tests/test_validate.py::test_validate_bids_errors[invalid_pet001]
========================================================= short test summary info ==========================================================
FAILED tests/test_validate.py::test_validate_bids_errors[invalid_asl003] - TypeError: object of type 'generator' has no len()
FAILED tests/test_validate.py::test_validate_bids_errors[invalid_eeg_cbm] - TypeError: object of type 'generator' has no len()
FAILED tests/test_validate.py::test_validate_bids_errors[invalid_pet001] - TypeError: object of type 'generator' has no len()
============================================================ 3 failed in 1.07s =============================================================
(dev) [deco]~/src/dandi-cli/dandi ❱ head -76 tests/test_validate.py | tail -5
    with open(os.path.join(selected_dataset, ".ERRORS.json")) as f:
        expected_errors = json.load(f)

    # We know that these datasets contain errors.
    assert len(validation_result) > 0
```

Reply:
So if we're just converting a generator to a list, write it as `validation_result = list(validation_result)` so it looks less ridiculous.